app/models/product_test.rb

```diff
@@ -30,7 +32,7 @@ class ProductTest
   mount_uploader :patient_archive, PatientArchiveUploader
 
   after_create do |product_test|
-    ProductTestSetupJob.perform_later(product_test)
+    ProductTestSetupJob.perform_later(product_test) unless product_test.is_a? FilteringTest
```

Comment:
I know why this is being done.  The filtering test used to build the tests, generate the  records and then save the tests.  This allowed the records to be created before the test was actually created and saved and allowed this callback to do its job.  Now however that is not the case and having this callback seems only meaningful for MeasureTest so maybe it should be moved there.